src/App.tsx

```diff
@@ -1,19 +1,166 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 import React from 'react';
 import './App.css';
 
-import users from './api/users';
+import { TodoList } from './components/TodoList';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
 
-const App: React.FC = () => {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+function getUserById(userId: number): User | null {
+  return usersFromServer.find(user => user.id === userId)
+      || null;
+}
+
+const TodosWithUser: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+type State = {
+  users: User[],
+  todos: Todo[],
+
+  hasTitleError: boolean,
+  title: string,
+
+  hasUserError: boolean,
+  newUserId: number,
 };
 
+class App extends React.PureComponent {
+  state: State = {
+    users: usersFromServer,
```

Comment:
no need to save usersFromServer into state, you don't change them, just use usersFromServer directly